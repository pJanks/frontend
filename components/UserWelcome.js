import * as React from 'react';
import { Text, View, StyleSheet } from 'react-native';

export default function UserWelcome() {
  let today = new Date();
  let dd = String(today.getDate()).padStart(2, '0');
  let mm = String(today.getMonth() + 1).padStart(2, '0'); //January is 0!
  let yyyy = today.getFullYear();

  today = yyyy + '/' + mm + '/' + dd;

  return (
    <View style={ styles.container }>
      <Text style={ styles.welcome }>Welcome, [Username]!</Text>
      <Text style={ styles.date }>Today's date is {today}</Text>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'lightblue',
    padding: 10,
  },
  welcome: {
    fontSize: 24,
    textAlign: 'center',
  },
  date: {
    fontSize: 16,
    textAlign: 'center',
  }
})

