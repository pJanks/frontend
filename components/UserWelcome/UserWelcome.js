import * as React from 'react';
import { Text, View, StyleSheet } from 'react-native';

export default function UserWelcome(userInfo) {
  let today = new Date();
  let dd = String(today.getDate()).padStart(2, '0');
  let mm = String(today.getMonth() + 1).padStart(2, '0'); //January is 0!
  let yyyy = today.getFullYear();

  today = yyyy + '/' + mm + '/' + dd;
  return (
    <View style={ styles.container }>
      <Text style={ styles.welcome }>Welcome, {`${userInfo.userInfo.firstName} ${userInfo.userInfo.lastName}`}!</Text>
      <Text style={ styles.date }>Today's date is {today}</Text>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#FFBD58',
    padding: 10,
  },
  welcome: {
    fontSize: 24,
    textAlign: 'center',
  },
  date: {
    fontSize: 16,
    textAlign: 'center',
  }
})
