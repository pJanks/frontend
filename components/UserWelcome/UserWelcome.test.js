import React from 'react';
import UserWelcome from './UserWelcome';

import renderer from 'react-test-renderer';

test('renders correctly', () => {

  const userWelcomePage = renderer.create(<UserWelcome />).toJSON();

  expect(userWelcomePage).toMatchSnapshot();
});
