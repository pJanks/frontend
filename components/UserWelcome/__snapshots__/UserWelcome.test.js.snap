// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFBD58",
      "padding": 10,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 24,
        "textAlign": "center",
      }
    }
  >
    Welcome, 
    undefined undefined
    !
  </Text>
  <Text
    style={
      Object {
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    Today's date is 
    2020/04/15
  </Text>
</View>
`;
